plugins/grid-bookmark/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { lazy } from 'react'
+
+import BookmarkIcon from '@material-ui/icons/Bookmark'
+import BookmarksIcon from '@material-ui/icons/Bookmarks'
+
+import WidgetType from '@jbrowse/core/pluggableElementTypes/WidgetType'
+import Plugin from '@jbrowse/core/Plugin'
+import PluginManager from '@jbrowse/core/PluginManager'
+import { LinearGenomeViewModel } from '@jbrowse/plugin-linear-genome-view'
+import { getSession } from '@jbrowse/core/util'
+import ViewType from '@jbrowse/core/pluggableElementTypes/ViewType'
+import { isSessionModelWithWidgets } from '@jbrowse/core/util'
+
+import {
+  stateModelFactory as GridBookmarkStateModelFactory,
+  configSchema as GridBookmarkConfigSchema,
+} from './GridBookmarkWidget'
+
+export default class extends Plugin {
+  name = 'GridBookmarkPlugin'
+
+  install(pluginManager: PluginManager) {
+    pluginManager.addWidgetType(() => {
+      return new WidgetType({
+        name: 'GridBookmarkWidget',
+        heading: 'Bookmarked regions',
+        configSchema: GridBookmarkConfigSchema,
+        stateModel: GridBookmarkStateModelFactory(pluginManager),
+        ReactComponent: lazy(
+          () => import('./GridBookmarkWidget/components/GridBookmarkWidget'),
+        ),
+      })
+    })
+
+    pluginManager.extendPluggable('LinearGenomeView', LinearGenomeView => {
+      const { stateModel } = LinearGenomeView as ViewType
+      const newStateModel = stateModel.extend((self: LinearGenomeViewModel) => {
+        const superMenuItems = self.menuItems
+        const superRubberBandMenuItems = self.rubberBandMenuItems
+        return {
+          actions: {
+            activateBookmarkWidget() {
+              const session = getSession(self)
+              if (isSessionModelWithWidgets(session)) {
+                // @ts-ignore
+                let bookmarkWidget = session.widgets.get('GridBookmark')
+                if (!bookmarkWidget) {
+                  bookmarkWidget = session.addWidget(
+                    'GridBookmarkWidget',
+                    'GridBookmark',
+                    { view: self },
+                  )
+                }
+
+                session.showWidget(bookmarkWidget)
+                return bookmarkWidget
+              }
+
+              throw new Error('Could not open bookmark widget')
+            },
+
+            bookmarkCurrentRegion() {
+              const selectedRegions = self.getSelectedRegions(
+                self.leftOffset,
+                self.rightOffset,
+              )
+              const firstRegion = selectedRegions[0]
+              // @ts-ignore
+              const { widgets } = getSession(self)
```

Comment:
Probably needs a `isSessionModelWithWidgets` check like above to get rid of ts-ignore